aiida/orm/importexport.py

```diff
@@ -977,14 +977,16 @@ def import_data_sqla(in_path, ignore_unknown_nodes=False, silent=False):
                         #                        for n in Model.objects.filter(
                         #     **{'{}__in'.format(unique_identifier):
                         #            import_unique_ids})}
+                        relevant_db_entries = set()
+                        if len(import_unique_ids) > 0:
+                            qb = QueryBuilder()
+                            qb.append(Model, filters={unique_identifier: {"in": import_unique_ids}},
+                                      project=["*"], tag="res")
+                            relevant_db_entries = {getattr(v[0], unique_identifier): v[0] for v in qb.all()}
```

Comment:
If I understand correctly, relevant_db_entries is a dictionary if import_unique_ids contains something, and a set if not?
That should be improved, so we have at least a defined type. Also, the code will break below if that variable continues storing a set! (see line 1019).